pkg/gadgets/tcpconnect/tracer/core/bpf/tcpconnect.bpf.c

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+//
+// Based on tcpconnect(8) from BCC by Brendan Gregg
+#include <vmlinux/vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include "maps.bpf.h"
+#include "tcpconnect.h"
+
+SEC(".rodata") int filter_ports[MAX_PORTS];
+const volatile int filter_ports_len = 0;
+const volatile uid_t filter_uid = -1;
+const volatile pid_t filter_pid = 0;
+const volatile bool do_count = 0;
+const volatile bool filter_by_mnt_ns = false;
+
+/* Define here, because there are conflicts with include files */
+#define AF_INET		2
+#define AF_INET6	10
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u32);
+	__type(value, struct sock *);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} sockets SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct ipv4_flow_key);
+	__type(value, u64);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} ipv4_count SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct ipv6_flow_key);
+	__type(value, u64);
+	__uint(map_flags, BPF_F_NO_PREALLOC);
+} ipv6_count SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(u32));
+	__uint(value_size, sizeof(u32));
+} events SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, 1024);
+	__uint(key_size, sizeof(u64));
+	__uint(value_size, sizeof(u32));
+} mount_ns_set SEC(".maps");
+
+
+static __always_inline bool filter_port(__u16 port)
+{
+	return false;
+// TODO: this code causes a verification problem.
```

Comment:
What is the particular problem?
Is it known upstream?

Reply:
The verifier output for this is longer than the program itself.. 

I don't think upstream is aware of this as it doesn't happen with libbpf. I tried to create a reproducer for cilium/ebpf but gave up. Will add to my TODO list. 

```
error creating tracer: Failed to load ebpf program: field TcpV4ConnectRet: program tcp_v4_connect_ret: load program: permission denied: trace type programs with run-time allocated hash maps are unsafe. Switch to preallocated hash maps.
trace type programs with run-time allocated hash maps are unsafe. Switch to preallocated hash maps.
trace type programs with run-time allocated hash maps are unsafe. Switch to preallocated hash maps.
trace type programs with run-time allocated hash maps are unsafe. Switch to preallocated hash maps.
trace type programs with run-time allocated hash maps are unsafe. Switch to preallocated hash maps.
Unrecognized arg#0 type PTR
; int BPF_KRETPROBE(tcp_v4_connect_ret, int ret)
0: (bf) r6 = r1
; int BPF_KRETPROBE(tcp_v4_connect_ret, int ret)
1: (79) r7 = *(u64 *)(r6 +80)
; __u64 pid_tgid = bpf_get_current_pid_tgid();
2: (85) call bpf_get_current_pid_tgid#14
3: (bf) r8 = r0
; __u32 tid = pid_tgid;
4: (63) *(u32 *)(r10 -84) = r8
5: (bf) r2 = r10
; 
6: (07) r2 += -84
; skpp = bpf_map_lookup_elem(&sockets, &tid);
7: (18) r1 = 0xffff8f430894b800
9: (85) call bpf_map_lookup_elem#1
; if (!skpp)
10: (15) if r0 == 0x0 goto pc+174
 R0=map_value(id=0,off=0,ks=4,vs=8,imm=0) R6=ctx(id=0,off=0,imm=0) R7=inv(id=0) R8=inv(id=1) R10=fp0 fp-88=mmmm????
; int BPF_KRETPROBE(tcp_v4_connect_ret, int ret)
11: (67) r7 <<= 32
12: (77) r7 >>= 32
; if (ret)
13: (55) if r7 != 0x0 goto pc+166
 R0=map_value(id=0,off=0,ks=4,vs=8,imm=0) R6=ctx(id=0,off=0,imm=0) R7_w=inv0 R8=inv(id=1) R10=fp0 fp-88=mmmm????
14: (b7) r1 = 12
; sk = *skpp;
15: (79) r7 = *(u64 *)(r0 +0)
 R0=map_value(id=0,off=0,ks=4,vs=8,imm=0) R1_w=inv12 R6=ctx(id=0,off=0,imm=0) R7_w=inv0 R8=inv(id=1) R10=fp0 fp-88=mmmm????
16: (bf) r3 = r7
17: (0f) r3 += r1
18: (bf) r1 = r10
; 
19: (07) r1 += -98
; BPF_CORE_READ_INTO(&dport, sk, __sk_common.skc_dport);
20: (b7) r2 = 2
21: (85) call bpf_probe_read_kernel#113
last_idx 21 first_idx 10
regs=4 stack=0 before 20: (b7) r2 = 2
; if (filter_port(dport))
22: (69) r1 = *(u16 *)(r10 -98)
; if (filter_ports_len == 0)
23: (18) r2 = 0xffff8f430894d110
25: (61) r3 = *(u32 *)(r2 +0)
 R0=inv(id=0) R1_w=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=map_value(id=0,off=0,ks=4,vs=280,imm=0) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (filter_ports_len == 0)
26: (15) if r3 == 0x0 goto pc+19
 R0=inv(id=0) R1_w=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=map_value(id=0,off=0,ks=4,vs=280,imm=0) R3_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
27: (61) r2 = *(u32 *)(r2 +0)
 R0=inv(id=0) R1_w=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=map_value(id=0,off=0,ks=4,vs=280,imm=0) R3_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
28: (67) r2 <<= 32
29: (c7) r2 s>>= 32
30: (b7) r3 = 1
; for (i = 0; i < filter_ports_len; i++) {
31: (6d) if r3 s> r2 goto pc+148
 R0=inv(id=0) R1_w=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv(id=0,umin_value=1,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R3_w=inv1 R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
32: (b7) r2 = 0
33: (18) r3 = 0xffff8f430894d120
35: (18) r4 = 0xffff8f430894d110
37: (05) goto pc+6
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv0 R3=map_value(id=0,off=16,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv0 R3=map_value(id=0,off=16,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv1 R3_w=map_value(id=0,off=20,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv1 R3_w=map_value(id=0,off=20,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=2,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv1 R3_w=map_value(id=0,off=20,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=2,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv1 R3_w=map_value(id=0,off=20,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv2 R3_w=map_value(id=0,off=24,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv2 R3=map_value(id=0,off=24,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5=inv(id=0,umin_value=3,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv2 R3=map_value(id=0,off=24,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=3,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv2 R3=map_value(id=0,off=24,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv3 R3_w=map_value(id=0,off=28,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv3 R3_w=map_value(id=0,off=28,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=4,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv3 R3_w=map_value(id=0,off=28,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=4,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv3 R3=map_value(id=0,off=28,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv4 R3_w=map_value(id=0,off=32,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv4 R3_w=map_value(id=0,off=32,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=5,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv4 R3_w=map_value(id=0,off=32,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=5,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv4 R3_w=map_value(id=0,off=32,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv5 R3_w=map_value(id=0,off=36,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv5 R3_w=map_value(id=0,off=36,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=6,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv5 R3_w=map_value(id=0,off=36,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=6,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv5 R3_w=map_value(id=0,off=36,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv6 R3_w=map_value(id=0,off=40,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv6 R3_w=map_value(id=0,off=40,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=7,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv6 R3_w=map_value(id=0,off=40,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=7,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv6 R3_w=map_value(id=0,off=40,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv7 R3_w=map_value(id=0,off=44,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv7 R3_w=map_value(id=0,off=44,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=8,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv7 R3_w=map_value(id=0,off=44,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=8,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv7 R3_w=map_value(id=0,off=44,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv8 R3_w=map_value(id=0,off=48,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv8 R3_w=map_value(id=0,off=48,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=9,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv8 R3_w=map_value(id=0,off=48,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=9,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv8 R3_w=map_value(id=0,off=48,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv9 R3_w=map_value(id=0,off=52,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv9 R3_w=map_value(id=0,off=52,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=10,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv9 R3_w=map_value(id=0,off=52,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=10,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv9 R3_w=map_value(id=0,off=52,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv10 R3_w=map_value(id=0,off=56,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv10 R3_w=map_value(id=0,off=56,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=11,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv10 R3_w=map_value(id=0,off=56,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=11,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv10 R3_w=map_value(id=0,off=56,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv11 R3_w=map_value(id=0,off=60,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv11 R3_w=map_value(id=0,off=60,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=12,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv11 R3_w=map_value(id=0,off=60,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=12,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv11 R3_w=map_value(id=0,off=60,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv12 R3_w=map_value(id=0,off=64,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv12 R3_w=map_value(id=0,off=64,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=13,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv12 R3_w=map_value(id=0,off=64,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=13,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv12 R3_w=map_value(id=0,off=64,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv13 R3_w=map_value(id=0,off=68,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv13 R3_w=map_value(id=0,off=68,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=14,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv13 R3=map_value(id=0,off=68,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=14,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv13 R3=map_value(id=0,off=68,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv14 R3_w=map_value(id=0,off=72,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv14 R3_w=map_value(id=0,off=72,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=15,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv14 R3_w=map_value(id=0,off=72,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=15,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv14 R3_w=map_value(id=0,off=72,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv15 R3_w=map_value(id=0,off=76,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv15 R3_w=map_value(id=0,off=76,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=16,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv15 R3_w=map_value(id=0,off=76,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=16,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv15 R3_w=map_value(id=0,off=76,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv16 R3_w=map_value(id=0,off=80,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv16 R3_w=map_value(id=0,off=80,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=17,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv16 R3_w=map_value(id=0,off=80,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=17,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv16 R3_w=map_value(id=0,off=80,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv17 R3_w=map_value(id=0,off=84,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv17 R3_w=map_value(id=0,off=84,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=18,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv17 R3_w=map_value(id=0,off=84,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=18,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv17 R3_w=map_value(id=0,off=84,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv18 R3_w=map_value(id=0,off=88,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv18 R3_w=map_value(id=0,off=88,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=19,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv18 R3_w=map_value(id=0,off=88,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=19,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv18 R3_w=map_value(id=0,off=88,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv19 R3_w=map_value(id=0,off=92,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv19 R3_w=map_value(id=0,off=92,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=20,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv19 R3_w=map_value(id=0,off=92,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=20,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv19 R3_w=map_value(id=0,off=92,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv20 R3_w=map_value(id=0,off=96,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv20 R3_w=map_value(id=0,off=96,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=21,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv20 R3_w=map_value(id=0,off=96,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=21,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv20 R3_w=map_value(id=0,off=96,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv21 R3_w=map_value(id=0,off=100,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv21 R3_w=map_value(id=0,off=100,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=22,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv21 R3_w=map_value(id=0,off=100,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=22,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv21 R3_w=map_value(id=0,off=100,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv22 R3_w=map_value(id=0,off=104,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv22 R3_w=map_value(id=0,off=104,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=23,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv22 R3_w=map_value(id=0,off=104,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=23,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv22 R3_w=map_value(id=0,off=104,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv23 R3_w=map_value(id=0,off=108,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv23 R3_w=map_value(id=0,off=108,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=24,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv23 R3=map_value(id=0,off=108,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=24,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv23 R3=map_value(id=0,off=108,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv24 R3_w=map_value(id=0,off=112,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv24 R3_w=map_value(id=0,off=112,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=25,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv24 R3_w=map_value(id=0,off=112,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=25,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv24 R3_w=map_value(id=0,off=112,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv25 R3_w=map_value(id=0,off=116,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv25 R3_w=map_value(id=0,off=116,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=26,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv25 R3_w=map_value(id=0,off=116,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=26,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv25 R3_w=map_value(id=0,off=116,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv26 R3_w=map_value(id=0,off=120,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv26 R3_w=map_value(id=0,off=120,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=27,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv26 R3_w=map_value(id=0,off=120,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=27,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv26 R3_w=map_value(id=0,off=120,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv27 R3_w=map_value(id=0,off=124,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv27 R3_w=map_value(id=0,off=124,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=28,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv27 R3_w=map_value(id=0,off=124,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=28,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv27 R3_w=map_value(id=0,off=124,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv28 R3_w=map_value(id=0,off=128,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv28 R3_w=map_value(id=0,off=128,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=29,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv28 R3_w=map_value(id=0,off=128,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=29,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv28 R3_w=map_value(id=0,off=128,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv29 R3_w=map_value(id=0,off=132,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv29 R3_w=map_value(id=0,off=132,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=30,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv29 R3_w=map_value(id=0,off=132,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=30,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv29 R3_w=map_value(id=0,off=132,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv30 R3_w=map_value(id=0,off=136,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv30 R3_w=map_value(id=0,off=136,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=31,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv30 R3_w=map_value(id=0,off=136,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=31,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv30 R3_w=map_value(id=0,off=136,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv31 R3_w=map_value(id=0,off=140,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv31 R3_w=map_value(id=0,off=140,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=32,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv31 R3_w=map_value(id=0,off=140,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=32,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv31 R3_w=map_value(id=0,off=140,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv32 R3_w=map_value(id=0,off=144,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv32 R3_w=map_value(id=0,off=144,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=33,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv32 R3_w=map_value(id=0,off=144,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=33,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv32 R3_w=map_value(id=0,off=144,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv33 R3_w=map_value(id=0,off=148,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv33 R3_w=map_value(id=0,off=148,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=34,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv33 R3=map_value(id=0,off=148,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=34,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2=inv33 R3=map_value(id=0,off=148,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv34 R3_w=map_value(id=0,off=152,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv34 R3_w=map_value(id=0,off=152,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=35,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv34 R3_w=map_value(id=0,off=152,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=35,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv34 R3_w=map_value(id=0,off=152,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv35 R3_w=map_value(id=0,off=156,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; for (i = 0; i < filter_ports_len; i++) {
41: (67) r5 <<= 32
42: (c7) r5 s>>= 32
; for (i = 0; i < filter_ports_len; i++) {
43: (7d) if r2 s>= r5 goto pc+136
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv35 R3_w=map_value(id=0,off=156,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=36,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
44: (61) r5 = *(u32 *)(r3 +0)
 R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv35 R3_w=map_value(id=0,off=156,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umin_value=36,umax_value=2147483647,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; if (port == filter_ports[i])
45: (5d) if r5 != r1 goto pc-8

from 45 to 38: R0=inv(id=0) R1=inv(id=0,umax_value=65535,var_off=(0x0; 0xffff)) R2_w=inv35 R3_w=map_value(id=0,off=156,ks=4,vs=280,imm=0) R4=map_value(id=0,off=0,ks=4,vs=280,imm=0) R5_w=inv(id=0,umax_value=4294967295,var_off=(0x0; 0xffffffff)) R6=ctx(id=0,off=0,imm=0) R7=inv(id=3) R8=inv(id=1) R10=fp0 fp-88=mmmm???? fp-104=mm??????
; 
38: (07) r2 += 1
; for (i = 0; i < filter_ports_len; i++) {
39: (07) r3 += 4
; for (i = 0; i < filter_ports_len; i++) {
40: (61) r5 = *(u32 *)(r4 +0)
 R0=i (truncated...)
```

Reply:
The log is indeed huge... I am sorry but I did not check it entirely.
Nonetheless, does it also occur removing the `__always_inline` attribute?

Reply:
Yes, it happens the same.

Reply:
Sad :(